src/components/MapComponent/index.tsx

```diff
@@ -3,16 +3,26 @@ import { Circle, color, create } from '@amcharts/amcharts4/core';
 import { MapChart, projections, MapPolygonSeries, MapImageSeries } from '@amcharts/amcharts4/maps';
 import am4geodata_worldLow from '@amcharts/amcharts4-geodata/worldLow';
 import MapComponentProps from './MapComponentProps';
-import { defaultModite } from '../../models/Modite';
+import Modite, { defaultModite } from '../../models/Modite';
 import s from './styles.module.css';
 
-function MapComponent({ modite = defaultModite }: MapComponentProps) {
+let map: MapChart;
+let imageSeries: any;
+
+function updateMap(map: MapChart, markerData: any) {
+  imageSeries.data = markerData;
+    if (markerData.length === 1) {
+      const { latitude, longitude } = markerData[0];
+      map.zoomToGeoPoint({ latitude, longitude }, 5, true, 500);
+    } else {
+      map.goHome(500);
+    }
+};
+
+function MapComponent({ modites = defaultModite }: MapComponentProps) {
   const mapRef: React.MutableRefObject<null> = useRef(null);
-  let map: MapChart;
 
   useEffect(() => {
```

Comment:
I would love to understand if the stuff in this block involve heavy processing and if we should move any of it to a web worker

Reply:
Organizing the map marker data?  Or the creation of the MapChart itself?  Other?

Reply:
anything works 😄